### PR TITLE
kwhperiod_render.js

### PR DESCRIPTION
**Pull request by danbates2**

### CODE REVIEW

**Comment by danbates2** [comment]
Not sure why some of the files have changed here.
I made an Auto-Format of the kwhperiod widget, so it looks like a larger diff than it is.

**Comment by danbates2** [comment]
I've rebased my fork. Should be good to go.

File: widget/kwhperiod/kwhperiod_render.js [hunk 1]
```diff
@@ -5,394 +5,849 @@
...
+    } else {
+      // thisperiodEndValue
+      /*
+      var result = feed.get_value(feedid, now.getTime())
+      console.log(result, 'recent period end / now')
+      var thisperiodEndValue = result[1]
+      */
+      // thisPeriodStartValue
+      var result = feed.get_value(feedid, thisPeriodStartTime.getTime())
+      console.log(result, 'recent period start value')
```
Comment (emrysr) [inline widget/kwhperiod/kwhperiod_render.js]: keep output clean in production. should only output to console during testing

File: widget/kwhperiod/kwhperiod_render.js [hunk 2]
```diff
@@ -5,394 +5,849 @@
...
+      var thisPeriodStartValue = result[1]
+
+      // ... and calculate the result of our time window.
+      console.log(val, 'value now')
```
Comment (emrysr) [inline widget/kwhperiod/kwhperiod_render.js]: keep output clean in production. should only output to console during testing

**Comment by emrysr** [comment]
do you use the same wiget to compare the previous and current periods??

> eg. showing today's accumulative total as well as yesterday's

I only see one dashboard widget `kwhperiod`...
delwedd

**Comment by danbates2** [comment]
I've commented out the console logs.
Re: question if this is the only widget - YES! It's capable of the Recent Period and Past Period according to the setting 'Periods Ago'.
Did you spot the documentation I made? Ideally I would like to like the documentation in the widget settings dialog box. Any idea how we might do this?

**Comment by danbates2** [comment]
I've found a bug in this. If the data is not available the result falls back on the most recent value of the feed, which is misleading. A Null should be displayed instead I believe.

**Comment by danbates2** [comment]
I believe the bug is now fixed, however it's not a very clever fix. It'll do.

**Comment by TrystanLea** [comment]
Thanks @danbates2!!

**Pull request merged**